extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -149,33 +171,33 @@ function createDicomWebApi(dicomWebConfig, userAuthenticationService) {
        * Generates a URL that can be used for direct retrieve of the bulkdata
        *
        * @param {object} params
-       * @param {string} params.tag is the tag name of the URL to retrieve
-       * @param {object} params.instance is the instance object that the tag is in
-       * @param {string} params.defaultType is the mime type of the response
-       * @param {string} params.singlepart is the type of the part to retrieve
        * @returns an absolute URL to the resource, if the absolute URL can be retrieved as singlepart,
        *    or is already retrieved, or a promise to a URL for such use if a BulkDataURI
        */
       directURL: params => {
+        const client = clientManager.getClient(params?.instance?.clientName);
         return getDirectURL(
           {
-            wadoRoot: dicomWebConfig.wadoRoot,
-            singlepart: dicomWebConfig.singlepart,
+            wadoRoot: client.wadoRoot,
+            singlepart: client.singlepart,
           },
           params
         );
       },
-      bulkDataURI: async ({ StudyInstanceUID, BulkDataURI }) => {
-        qidoDicomWebClient.headers = getAuthrorizationHeader();
+      bulkDataURI: async ({ StudyInstanceUID, BulkDataURI, clientName }) => {
+        clientManager.setQidoHeaders();
```

Comment:
You should move this down to the `getQidoClient` similar to others that you set the headers yourself